misk/src/main/kotlin/misk/security/authz/FakeCallerAuthenticator.kt

```diff
@@ -40,14 +42,4 @@ class FakeCallerAuthenticator @Inject constructor(
 
 @Qualifier
 @Target(AnnotationTarget.FIELD, AnnotationTarget.FUNCTION)
-annotation class DevelopmentOnly
-
-/**
- * https://github.com/google/guice/wiki/FrequentlyAskedQuestions#how-can-i-inject-optional-parameters-into-a-constructor
- */
-@Singleton
-class OptionalBinder {
-  @com.google.inject.Inject(optional = true)
-  @DevelopmentOnly
-  var developmentCaller: MiskCaller? = null
```

Comment:
Yeah, the API hasn't changed. The `OptionalBinder` pattern lets you pass an optional injection into a constructor. I moved it to be an optional injection on a field instead, which is what `OptionalBinder` does itself. In misk, there's no benefit to do constructor optional injection, since we use Guice to bind tests too.